fsspec/tests/conftest.py

```diff
@@ -173,3 +175,45 @@ def serve():
 def server():
     with serve() as s:
         yield s
+
+
+def _populate_test_dir(path):
+    path.joinpath("file1").write_text("file1")
+    path.joinpath("file2").write_text("file2")
+    f1 = path.joinpath("folder1")
+    f1.mkdir()
+    f2 = path.joinpath("folder2")
+    f2.mkdir()
+    f1.joinpath("file3.txt").write_text("file3")
+    f2.joinpath("file4.txt").write_text("file4")
+
+
+@pytest.fixture(scope="session")
+def stdlib_simple_http_server(tmp_path_factory):
```

Comment:
Why do we need a different server implementation?

Reply:
I initially found the issue in tests with the stdlib `http.server` module, and needed a nested folder to reproduce. I'll check again if I can use/extend the existing test server.